src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,179 @@
+    }, {});
+
+  const onClickOnSender = (sender: WithId<User>) => {
+    const chatsToUpdate = privateChats.filter(
+      (chat) => !chat.isRead && chat.from === sender.id
+    );
+    chatsToUpdate.map(
+      (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+    );
+    setSelectedUser(sender);
+  };
+
+  const roundToNearestHour = (seconds: number) => {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  };
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = async (data: { messageToTheBand: string }) => {
+    chatContext &&
+      user &&
+      chatContext.sendPrivateChat(
+        user.uid,
+        selectedUser!.id,
+        data.messageToTheBand
+      );
+  };
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
+      />
+      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
+    </>
+  ) : (
+    <>
+      {hasPrivateChats && (
+        <div className="private-container show">
+          <div className="private-messages-list">
+            {Object.keys(discussionPartnerWithLastMessageExchanged)
+              .sort((a, b) =>
+                discussionPartnerWithLastMessageExchanged[b].ts_utc
+                  .valueOf()
+                  .localeCompare(
+                    discussionPartnerWithLastMessageExchanged[
+                      a
+                    ].ts_utc.valueOf()
+                  )
+              )
+              .map((userId: string) => {
+                const sender = { ...chatUsers![userId], id: userId };
+                const lastMessageExchanged =
+                  discussionPartnerWithLastMessageExchanged?.[userId];
+                return (
+                  <div
+                    key={userId}
+                    className="private-message-item"
+                    onClick={() => onClickOnSender(sender)}
+                    id="private-chat-modal-select-private-recipient"
+                  >
+                    <UserProfilePicture
+                      avatarClassName="private-message-author-pic"
+                      user={sender}
+                      setSelectedUserProfile={() => null}
```

Comment:
As above RE: `useCallback`

Reply:
still an issue in 77fc5d6d1829a42da2152d00871cde0b6f47269d, added new comment there, resolving this